test/parallel/test-http-client-timeout-option.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var http = require('http');
+
+var options = {
+  method: 'GET',
+  port: undefined,
+  host: '127.0.0.1',
+  path: '/',
+  timeout: 1
+};
+
+var server = http.createServer(function(req, res) {
+  // this space intentionally left blank
+});
+
+server.listen(0, options.host, function() {
+  options.port = this.address().port;
+  var req = http.request(options, function(res) {
+    // this space intentionally left blank
+  });
+  req.on('error', function() {
+    // this space is intentionally left blank
```

Comment:
It does throw a `socket hang up` error. I've checked other timeout related tests and they do that as well, so I guess it's ok.